samsungctl/__main__.py

```diff
@@ -96,18 +99,22 @@ def main():
         return
 
     try:
-        with Remote(config) as remote:
-            for key in args.key:
-                remote.control(key)
-
-            if args.interactive:
-                logging.getLogger().setLevel(logging.ERROR)
-                interactive.run(remote)
-            elif len(args.key) == 0:
-                logging.warning("Warning: No keys specified.")
-    except Remote.ConnectionClosed:
+        if config["method"] == "legacy":
+            remote = RemoteLegacy(config)
+        elif config["method"] == "websocket":
+            remote = RemoteWebsocket(config)
```

Comment:
Fail here with an error message if the method is something else.

E.g. "Error: Unknown method 'asdf'".